src/ray/gcs/gcs_server/observable_internal_kv.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/gcs/gcs_server/observable_internal_kv.h"
+
+#include "absl/time/time.h"
+#include "ray/stats/metric_defs.h"
+
+namespace ray {
+namespace gcs {
+
+using namespace ray::stats;
+
+ObservableInternalKv::ObservableInternalKv(std::unique_ptr<InternalKVInterface> delegate)
+    : delegate_(std::move(delegate)){};
+
+void ObservableInternalKv::Get(const std::string &ns,
```

Comment:
Should we enable this only when non `memory` storage is used (since it will be mostly just event loop queueing delay in this case)?